engine/table/src/main/java/io/deephaven/engine/table/impl/BaseTable.java

```diff
@@ -525,12 +525,23 @@ public void awaitUpdate() throws InterruptedException {
     }
 
     @Override
-    public boolean awaitUpdate(long timeout) throws InterruptedException {
-        final MutableBoolean result = new MutableBoolean(false);
-        updateGraph.exclusiveLock().doLocked(
-                () -> result.setValue(ensureCondition().await(timeout, TimeUnit.MILLISECONDS)));
+    public boolean awaitUpdate(long timeoutMillis) throws InterruptedException {
 
-        return result.booleanValue();
+        final long startTime = System.nanoTime();
+        if (!updateGraph.exclusiveLock().tryLock(timeoutMillis, TimeUnit.MILLISECONDS)) {
+            // Usually, users will already be holding the exclusive lock when calling this method. If they are not and
+            // cannot acquire the lock within the timeout, we should return false now.
+            return false;
+        }
+        timeoutMillis -= TimeUnit.MILLISECONDS.convert(System.nanoTime() - startTime, TimeUnit.NANOSECONDS);
+
```

Comment:
I would check for timeoutMillis <= 0 and bail (or clamp it).  I see no javadoc for the await call on Condition, but the standard wait call has the note:
In all respects, this method behaves as if wait(0L, 0) had been called. See the specification of the [wait(long, int)](https://docs.oracle.com/javase/10/docs/api/java/lang/Object.html#wait(long,int)) method for details.

That has bit some of the Enterprise gRPC timeout handling in the past.

Which points out that if you have a zero argument, you are going to wait forever.  I would prefer to have a check within the try to be very explicit about that edge case.